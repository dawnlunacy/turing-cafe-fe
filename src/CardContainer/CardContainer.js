import React from 'react';
import Card from '../Card/Card';
import './CardContainer.css';



const CardContainer = (props) => {
    const reservations = props.reservations.map(reservation => {
            return ( <Card 
                key = { reservation.id }
                id={ reservation.id }
                name = { reservation.name }
                date = { reservation.date }
                time = { reservation.time }
                number = { reservation.number }
                /> )
        })  

    return (
        <section className="res-container">
            { reservations }
        </section>
    )
    
}

export default CardContainer;
