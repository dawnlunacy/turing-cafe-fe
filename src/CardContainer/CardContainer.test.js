import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer', () => {
  it('should return reservations cards matching snapshot', () => {
    const mockReservations = [
        {
            "id": 1,
            "name": "Christie",
            "date": "12/29",
            "time": "7:00",
            "number": 12
        },
        {
            "id": 2,
            "name": "Leta",
            "date": "4/5",
            "time": "7:00",
            "number": 2
        },
        {
            "id": 3,
            "name": "Pam",
            "date": "1/21",
            "time": "6:00",
            "number": 4
        }]
    const wrapper = shallow(
      <CardContainer reservations={mockReservations}/>
    )
    
    expect(wrapper).toMatchSnapshot();
  })
})