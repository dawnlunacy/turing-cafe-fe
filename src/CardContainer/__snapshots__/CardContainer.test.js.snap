// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should return reservations cards matching snapshot 1`] = `
<section
  className="res-container"
>
  <Card
    date="12/29"
    id={1}
    key="1"
    name="Christie"
    number={12}
    time="7:00"
  />
  <Card
    date="4/5"
    id={2}
    key="2"
    name="Leta"
    number={2}
    time="7:00"
  />
  <Card
    date="1/21"
    id={3}
    key="3"
    name="Pam"
    number={4}
    time="6:00"
  />
</section>
`;
